### ISSUE TITLE
DexOverflow issue firestore

### ISSUE THREAD

**Issue by DavidKuennen** [invertase/react-native-firebase#475]
I know this is propably the wrong place to ask this, but I have a problem after adding `compile "com.google.firebase:firebase-firestore:11.4.2"` to my build. 

As soon as I add that, it also adds `com.google.common` among other things to my dex file, which is around 27k extra references, thus bursting through the 64k dex limit.

Does anyone know why that is or am I doing something wrong?

**Comment by brianhiss** [comment]
I can confirm the same issue. I have `react-native-firebase 3.0.1` installed, with all Firebase Services running, when I try to add in `"com.google.firebase:firebase-firestore:11.4.2"`, I can dex error like @DavidKuennen.

![screen shot 2017-10-07 at 10 28 42 am](https://user-images.githubusercontent.com/1640257/31305811-d5bbfc9c-ab4a-11e7-8a4c-89bc4a314e5e.png)

**Comment by mjmaix** [comment]
same for me when I tried with all firebase services with `v11.4.2`

I have to add `multiDexEnabled true` and `dexOptions { javaMaxHeapSize "4g" }`

https://github.com/mjmaix/RNFirebase3/commit/e871aebfa3bfcf248209c61d79bde9bc5b756ef2

Would wan't to hear if there would be any problems from adding `multiDexEnabled true` or are there any other solution?

**Comment by brianhiss** [comment]
I can confirm the solution provided by @mjmaix works for me. 

I'm able to now build with `"com.google.firebase:firebase-firestore:11.4.2"`, but I am also unclear if changing the setting `multiDexEnabled true` leads to any other issues.

**Comment by DavidKuennen** [comment]
Seems like they are aware of the problem and are currently trying to reduce the sdk size: https://stackoverflow.com/questions/46614405/dexoverflowexception-after-adding-com-google-firebasefirebase-firestore11-4-2

I set it temporarily to multiDexEnabled (and changing Application to MultiDexApplication) and it's working again. Certainly not the solution I want, but I guess I'll just wait till they fix the problem so I can remove multiDex again.

I also tried proguarding, but it didn't help.

Closing this, since it really isn't any issue with react-native-firebase (which is doing an awesome job already providing the possibility to use the library with react native 👍 )
